benchmarks/benchmarker.py

```diff
@@ -0,0 +1,329 @@
+def main_compare(args):
+    rt = args.rev_this
+    ro = args.rev_other
+    repo = git.Repo(search_parent_directories=True)
+    rev_this = str(repo.commit(rt))
+    rev_other = str(repo.commit(ro))
+
+    filename = get_report_filename(args.filename)
+
+    with open(filename, 'r') as infile:
+        data = json.load(infile)
+
+    rev_this_benchmark = data[rev_this]
+    rev_other_benchmark = data[rev_other]
+
+    slowers = []
+    fasters = []
+    sames = []
+
+    for this_res in rev_this_benchmark:
+        for other_res in rev_other_benchmark:
+            if this_res["name"] == other_res["name"] \
+                    and this_res["params"] == other_res["params"]:
+                print(benchmark_desc(this_res["name"],
+                                     this_res["params"]))
+                print("\nShowing runtime "
+                      "{:6.6} ({:6.6}) / "
+                      "{:6.6} ({:6.6})".format(ro, rev_other,
+                                               rt, rev_this))
+                print("")
+                for N in this_res["Ns"]:
+                    N = str(N)
+                    this_t = this_res["size_scale"][N]
+                    other_t = other_res["size_scale"][N]
+                    ratio = other_t/this_t
+
+                    print("N: {}, ratio: {:0.2f}".format(N, ratio))
+
+                    info = {"name": this_res["name"],
+                            "params": this_res["params"],
+                            "N": N,
+                            "ratio": ratio}
+
+                    if ratio < 1:
+                        print("\t{:6.6} is {:0.2f} times "
+                              "slower than {:6.6}".format(rt, ratio, ro))
+                        slowers.append(info)
+                    if ratio > 1:
+                        print("\t{:6.6} is {:0.2f} times "
+                              "faster than {:6.6}".format(rt, ratio, ro))
+                        fasters.append(info)
+                    if ratio == 1:
+                        print("\t{:6.6} and {:6.6} "
+                              "have the same speed".format(rt, ro))
+                        sames.append(info)
+
+                num_threads = len(this_res["thread_scale"]) - 1
+                for i in range(1, num_threads + 1):
+                    for j, N in enumerate(this_res["Ns"]):
+                        this_t = this_res["thread_scale"][i][j]
+                        other_t = other_res["thread_scale"][i][j]
+                        ratio = other_t/this_t
+
+                        print("Threads: {}, N: {}, "
+                              "ratio: {:0.2f}".format(str(i), N, ratio))
+
+                        info = {"name": this_res["name"],
+                                "params": this_res["params"],
+                                "threads": i,
+                                "N": N,
+                                "ratio": ratio}
+
+                        if ratio < 1:
```

Comment:
Is there a way you could write this and the printing logic above as a single function?